soci/soci_index.go

```diff
@@ -94,6 +112,60 @@ type IndexDescriptorInfo struct {
 	CreatedAt time.Time
 }
 
+// DecodeIndex deserializes a JSON blob in an io.Reader
+// into a SOCI index. The blob can be either an OCI 1.1 Artifact
+// or an OCI 1.0 Manifest
+func DecodeIndex(r io.Reader, index *Index) error {
```

Comment:
any interest in `(i *Index) func DecodeIndex(r io.Reader) error`?

Reply:
I think `Index{}.DecodeIndex(r)` is less idiomatic in go than
```
var index Index
Unmarshal(r, &index)
```

The latter at least matches the `json` package. Maybe `Unmarshall` should return a `*Index` instead like `ztoc.Unmarshal` does?